packages/run-protocol/src/vpool-xyk-amm/stableSwapCurve.js

```diff
@@ -0,0 +1,299 @@
+// @ts-check
+import { assert, details as X } from '@agoric/assert';
+import { AmountMath } from '@agoric/ertp';
+import {
+  makeRatio,
+  floorMultiplyBy,
+  makeRatioFromAmounts,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { natSafeMath } from '../../../zoe/src/contractSupport/index.js';
+
+const { floorDivide } = natSafeMath;
+const A = 85n;
+const BASIS_POINTS = 10000n;
+const MAX_LOOP_LIMIT = 1000;
+
+/**
+ * Common functionality in getInputAmount and getOuptutAmount
+ * @param {Amount} inputAmount - the Amount of the asset sent
+ * in to be swapped.
+ * @param {Amount[]} poolAmounts - Array of amounts of each asset.Which is
+ *                                passed from the contract.
+ * @param {number} tokenIndexFrom - index of amount of inputReserve
+ *                                  in the reserves array.
+ * @param {number} tokenIndexTo - index of amount of outputReserve
+ *                                in the reserves array.
+ * @param {bigint} [feeBasisPoints=30n] - the fee taken in
+ *                                        in exchange
+ * @returns {{inputAmountAfterFeeCut:Amount, inputAmountWithoutFeeCut:Amount, poolValues: bigint[],poolBrands:Brand[] }} output - recomputed pool values
+ */
+const commonStablePrice = (
+  inputAmount,
+  poolAmounts,
+  tokenIndexFrom,
+  tokenIndexTo,
+  feeBasisPoints = 30n,
+) => {
+  const inputReserve = poolAmounts[tokenIndexFrom];
+  const outputReserve = poolAmounts[tokenIndexTo];
+  assert(
+    inputAmount.value > 0n,
+    X`inputValue ${inputAmount.value} must be positive`,
+  );
+  assert(
+    inputReserve.value > 0n,
+    X`inputReserve ${inputReserve.value} must be positive`,
+  );
+  assert(
+    outputReserve.value > 0n,
+    X`outputReserve ${outputReserve.value} must be positive`,
+  );
+  // Fee ratio calculation
+  const feeCutRatio = makeRatio(
+    BASIS_POINTS - feeBasisPoints,
+    inputAmount.brand,
+    BASIS_POINTS,
+    inputAmount.brand,
+  );
+  // Fee ratio multiplied by inputAmount to get inputAmount After fee cut
+  let inputAmountAfterFeeCut = floorMultiplyBy(inputAmount, feeCutRatio);
+  // Normalizing input amount according to pool value
+  inputAmountAfterFeeCut = AmountMath.make(
+    inputAmountAfterFeeCut.brand,
+    inputAmountAfterFeeCut.value,
+  );
```

Comment:
Using the same variable below in `line 61` so can't make it const.

Reply:
Line 61 makes an Amount from the components of an Amount. It's completely unnecessary. The value computed on line 59 (and reused on line 51) is all that's needed, so it can be `const`.